activejob/test/cases/exceptions_test.rb

```diff
@@ -370,6 +370,29 @@ class ExceptionsTest < ActiveSupport::TestCase
     assert_equal expected_array, JobBuffer.values.last(2)
   end
 
+  test "wait: :exponentially_longer is deprecated but still works" do
+    assert_deprecated(ActiveJob.deprecator) do
+      class ::RetryJob
+        retry_on LegacyExponentialNamingError, wait: :exponentially_longer, attempts: 10, jitter: nil
+      end
```

Comment:
Strictly speaking, reopening the class here to call `retry_on` means that this test has a side effect, but it is a convenient way of using `assert_deprecated`, and I think it's fairly low-risk since this callback is really specific to this test.
I'm almost hesitant to move the definition of `LegacyExponentialNamingError` here so that all the setup for this test is local to this file, but I might be overthinking it.

Reply:
Could you subclass the job instead with an anonymous class? (I haven't tried it so this may not work but I believe it should)

```
klass = Class.new(RetryJob) do
  retry_on ...
end
```

Reply:
Yeah, subclassing is a good idea. An anonymous class is a bit tricky just to get the `klass` variable outside of the `assert_deprecated` bock's scope for calling `klass.perform_later`, so I ended up just naming the subclass.
I also moved the error definition into this file while I was at it.

Reply:
ah you're right... one more suggestion:

```
klass = Class.new(RetryJob)

assert_deprecated do
  klass.retry_on ...
end
```

Reply:
that's clever, and it almost works, but ActiveJob doesn't allow anonymous classes: I get this error when trying this implementation:
```
NoMethodError: undefined method `constantize' for nil:NilClass
    /Users/victormours/dev/rails/activejob/lib/active_job/core.rb:63:in `deserialize'
```

The line raising the error is `job = job_data["job_class"].constantize.new`, so the anonymous class can't be deserialized.

Reply:
Thanks for trying it out, sorry it didn't quite work 😅

Reply:
No worries! it was fun to tinker around a bit 😄